assertion_test.go

```diff
@@ -477,3 +477,183 @@ func TestAssertion_Strings(t *testing.T) {
+				Actual: &AssertionValue{},
+			},
+			traits: fieldTraits{
+				Actual: fieldDenied,
+			},
+		},
+		{
+			testName:          "required Expected",
+			errorContainsText: "should have Expected",
+			failure: AssertionFailure{
+				Expected: nil,
+			},
+			traits: fieldTraits{
+				Expected: fieldRequired,
+			},
+		},
+		{
+			testName:          "denied Expected",
+			errorContainsText: "can't have Expected",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{},
+			},
+			traits: fieldTraits{
+				Expected: fieldDenied,
+			},
+		},
+		{
+			testName:          "Required value on Expected to be pointer to AssertionRange",
+			errorContainsText: "AssertionValue should contain AssertionRange",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionRange{},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected to have min on range",
+			errorContainsText: "AssertionRange value should have non-nil Min field",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionRange{
+						Max: 1,
+					},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected to have max on range",
+			errorContainsText: "AssertionRange value should have non-nil Max field",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionRange{
+						Min: 1,
+					},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected to be pointer to AssertionList",
+			errorContainsText: "AssertionValue should contain AssertionList",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected to be pointer to AssertionList",
+			errorContainsText: "AssertionValue should contain AssertionList",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected AssertionList with len",
+			errorContainsText: "AssertionList should be non-empty",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required Expected AssertionList with values not being a slice",
+			errorContainsText: "but it contains a single element which itself is a list",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionList{[]int{1}},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required Expected",
```

Comment:
Let's rename this to "required List", because we already have "required Expected" above.